client/my-sites/email/email-management/home/email-plan/index.jsx

```diff
@@ -121,19 +123,72 @@ function EmailPlan( { domain, hideHeaderCake = false, selectedSite, source } ) {
 		);
 	};
 
-	const addEmailForwardMutationActive = useAddEmailForwardMutationIsLoading();
+	// Email checkout provides the /mailbox route with a new email param, e.g. /mailboxes/example.com?new-email=example@example.com
+	const queryParams = new URLSearchParams( window.location.search );
+	const newEmail = queryParams.get( 'new-email' );
+
+	const refetchTimeoutRef = useRef( null );
+	const [ refetchCount, setRefetchCount ] = useState( 0 );
 
-	const { data: emailAccounts = [], isLoading } = useGetEmailAccountsQuery(
-		selectedSite.ID,
-		domain.name,
-		{
-			refetchOnMount: ! addEmailForwardMutationActive,
-			retry: false,
+	useEffect( () => {
+		if ( ! newEmail ) {
+			return;
 		}
-	);
+		dispatch(
+			successNotice( translate( 'Your mailbox has been created.' ), {
```

Comment:
```suggestion
			successNotice( translate( 'Your mailbox %(mailbox)s has been created.', { args: { mailbox: newEmail } } ), {
```
Nit, but I think it'd be nice to mention the email address as a reassurance to users.